### PR TITLE
changing library colors and accepting color names

### PR DESCRIPTION
**Pull request by 2is10**
@ahsu1230

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 1]
```diff
@@ -247,36 +247,55 @@ object BasicLibraryStatistics {
   implicit val format = Json.format[BasicLibraryStatistics]
 }
 
-sealed abstract class HexColor(val hex: String)
-object HexColor {
+sealed abstract class LibraryColor(val hex: String)
+object LibraryColor {
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:251]: renaming to `LibraryColor` to better reflect that it cannot represent an arbitrary hex color

File: kifi-backend/modules/common/app/com/keepit/model/Library.scala [hunk 2]
```diff
@@ -247,36 +247,55 @@ object BasicLibraryStatistics {
...
       case PURPLE.hex => PURPLE
+
+      // TODO: Remove mappings below when migration from old colors is complete.
+      case "#3975bf" => BLUE
+      case "#fab200" => SKY_BLUE // was ORANGE_YELLOW
+      case "#2ec89a" => GREEN
+      case "#ff9430" => ORANGE
+      case "#e35957" => RED
+      case "#c764a2" => MAGENTA
+      case "#955cb4" => PURPLE
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:289]: I’ll prepare a database migration endpoint in a separate commit.

Comment (ahsu1230) [inline kifi-backend/modules/common/app/com/keepit/model/Library.scala:289]: nice!

**Comment by ahsu1230** [comment]
:+1:
 thanks

**Pull request merged**